tensorflow/python/keras/callbacks.py

```diff
@@ -443,6 +449,38 @@ class Callback(object):
           the number of samples in the current batch.
       on_batch_end: logs include `loss`, and optionally `acc`
           (if accuracy monitoring is enabled).
+    
+  Example:
+  ```python  
+  # Import the datetime module to display the time
+  import datetime
+  
+  # A custom callback displaying the loss and accuracy 
+  # from the `logs` dict at the end of a batch/epoch
+  class MyLossAccuracyCallback(tf.keras.callbacks.Callback):
+  
+    def on_train_batch_end(self, batch, logs=None):
+      print('For batch {}, loss = {:7.2f} at '.format(batch, logs['loss']), 
+            datetime.datetime.now().time())
+    
+    def on_test_batch_end(self, batch, logs=None):
+      print('For batch {}, loss = {:7.2f} at '.format(batch, logs['loss']),
+            datetime.datetime.now().time())
+            
+    def on_epoch_end(self, epoch, logs=None):
+      print('Average loss for epoch {} = {:7.2f}, accuracy = {:7.2f}.'
+            .format(epoch, logs['loss'], logs['acc']))
+  
+  batch_size = 128
+  epochs = 10
+  
+  model.fit(data, labels, 
+            batch_size=batch_size,
+            steps_per_epoch=int(np.ceil(X_train.shape[0])/batch_size),
```

Comment:
By x_train did you mean data?
Also, curious, what is the reason why we need np.ceil here?

Reply:
Just saw this @rchao:

1) Replacing `X_train` to `data` - habits!

2) Used `np.ceil` to be in line with @fchollet 's example [here](https://keras.io/examples/mnist_tfrecord/) on www.keras.io:
```python
train_model.fit(
    epochs=epochs,
    steps_per_epoch=int(np.ceil(data.train.num_examples / float(batch_size))),
    callbacks=[EvaluateInputTensor(test_model, steps=100)])
```